### PR TITLE
Updated Button Text for Navigation to Review Page

### PR DESCRIPTION
**Pull request by Erikang20**
### Description

When a user is filling out an application, they may want to return to the review page. This update changes the button text from "Save and finish application later" to "Review and exit" for clearer guidance.

**Before:**
Screenshot 2024-10-01 at 3 24 38 PM

**After**
Screenshot 2024-10-01 at 3 24 23 PM

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [X] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ X] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).

#### User visible changes

- [ X] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [ X] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ X] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [X ] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order
- [x] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

### Issue(s) this completes:

Fixes issue #8641

### CODE REVIEW

File: server/app/services/MessageKey.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public enum MessageKey {
   BUTTON_PREVIOUS_SCREEN("button.previousScreen"),
   BUTTON_PRINT("button.print"), // North Star only
   BUTTON_BACK("button.back"),
-  BUTTON_SAVE_AND_EXIT("button.saveAndExit"),
+  BUTTON_REVIEW_AND_EXIT_V2("button.reviewAndExit.v2"),
```
Comment (devin-civi) [inline server/app/services/MessageKey.java:108]: There isn't another reviewAndExit, right? So no need for the v2 here?

Comment (Erikang20) [reply server/app/services/MessageKey.java:108]: Correct. I'll remove the v2. 👍

**Review by ssandbekkhaug — APPROVED** [review]
Looks good to me, just resolve the merge conflict

**Review by devin-civi — APPROVED** [review]

**Comment by Erikang20** [comment]
Opening a new PR for this: https://github.com/civiform/civiform/pull/8838

**Pull request closed without merging**